packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -627,6 +548,31 @@ export function TransferPanel() {
     }
   }
 
+  const depositRequiresChainSwitch = () => {
```

Comment:
I don't think this is right. We're only checking for network layer here, so if you are connected to Sepolia but choose Ethereum, it will return true.

I think we never hit this case because it's used for analytics after the network is switched, but the method logic is still incorrect. If we reuse the method somewhere, it could cause bugs.

We could make this assumption before because it was within the transfer method, but now we have a potentially reusable method.

Same for `withdrawalRequiresChainSwitch`

Reply:
Hmm, I haven't changed the logic; just abstracted the previous logic into readable function blocks. These are just functions for checking the validity of connected source-destination chain pairs, before this, the correct network has already been switched to.

Would it help if these functions are moved inside the transfer function again so that they can't be reused anywhere else? Updated it, pls lemme know if it resolves your concern.